### PR TITLE
run-build.sh: git clean + git checkout -f

### PR DESCRIPTION
**Pull request by ghost**
git checkout may fail if a file in the current branch changed from being
tracked by git to being generated instead. To really clean the current
branch git clean should be used instead of git reset because it will get
rid of generated file. git reset will not remove generated files.

Add a --force to checkout because we really want to ignore whatever
safeguard git has in place to prevent file loss. Whatever leftovers
there are despite our precautions, we want them gone.

Signed-off-by: Loic Dachary loic@dachary.org

### CODE REVIEW

**Comment by ghost** [comment]
@tchaikov how does that look ?

File: run-build.sh [hunk 1]
```diff
@@ -30,10 +30,8 @@ _run()
 	for sub in `git submodule -q foreach pwd`; do
 	    rm -rf $sub
 	done
-	git reset --hard HEAD  # in case there were modified files
-	git checkout "$commit" &&
-	git reset --hard $commit || 
-	git reset --hard $commit || return 20
+        git clean -q -f -x -d
+	git checkout --force "$commit" || return 20
```
Comment (tchaikov) [inline run-build.sh:34]: bad indent?

Comment (ghost) [inline run-build.sh:34]: bad display rather

**Comment by tchaikov** [comment]
interesting... needs to rebase ....

**Comment by tchaikov** [comment]
:+1:

**Comment by ghost** [comment]
@tchaikov I was not modifying master. And I removed the Cleanup section which was modified to add rm -fr out~

**Pull request merged**